### PR TITLE
Users/subraman/iismainmodulerefactor

### PR DESCRIPTION
**Pull request by mvvsubbu**

### CODE REVIEW

File: Tasks/IISWebAppDeployment/Strings/resources.resjson/en-US/resources.resjson [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "loc.friendlyName": "IIS Web Application Deployment",
+  "loc.friendlyName": "Extension IIS Web Application Deployment",
```
Comment (mvvsubbu) [inline Tasks/IISWebAppDeployment/Strings/resources.resjson/en-US/resources.resjson]: This is place holder will change after discussing with PM

File: Tasks/IISWebAppDeployment/DeployIISWebApp.ps1 [hunk 2]
```diff
@@ -1,4 +1,179 @@
...
+        [string]$hostNameWithHttp,
+        [string]$hostNameWithSNI,
+        [string]$hostNameWithOutSNI,
+        [string]$sni
+    )
+    $hostName = [string]::Empty
+
+    if($protocol -eq "http")
+    {
+        $hostName = $hostNameWithHttp
```
Comment (kmkumaran) [inline Tasks/IISWebAppDeployment/DeployIISWebApp.ps1]: does this include both http/https?

**Comment by kmkumaran** [comment]
looks fine.

File: Tasks/IISWebAppDeployment/DeployIISWebApp.ps1 (hunk 2 above)
Comment (mvvsubbu) [inline Tasks/IISWebAppDeployment/DeployIISWebApp.ps1]: actually if protocol is not http and that means it is https. In this case if sni is enabled hostname is mandatory to be able to achieve that we have add two parameters in json, so appropriately reading the hostname input from UI

**Pull request merged**